pkg/diff/diff.go

```diff
@@ -71,6 +75,19 @@ func Diff(config, live *unstructured.Unstructured, opts ...Option) (*DiffResult,
 		live = remarshal(live, o)
 		Normalize(live, opts...)
 	}
+	// structuredMergeDiff is mainly used as a feature flag to enable
+	// calculating diffs using the structured-merge-diff library
+	// used in k8s while performing server-side applies. It checks the
+	// given diff Option or if the desired state resource has the
+	// Server-Side apply sync option annotation enabled.
+	structuredMergeDiff := o.structuredMergeDiff || resource.HasAnnotationOption(config, "argocd.argoproj.io/sync-options", "ServerSideApply=true")
```

Comment:
I tried to use `common.AnnotationSyncOptions` and `common.SyncOptionServerSideApply` instead of hardcoding the constants here. Unfortunately there is a (hard to solve) cyclic dependency with the `kube` package  if a do so. Im leaving it hardcoded for now but we should maybe refactor the `common` package to remove the `kube` dependency from it.